src/OpenCL.cpp

```diff
@@ -448,16 +453,22 @@ void OpenCL_Network::forward(const std::vector<net_t>& input,
                       conv2_weights,
                       &residualBuffer,
                       bn2_weights,
-                      true, skip_next_in_trans, true);
+                      true, skip_next_in_trans, true, !skip_next_in_trans);
             skip_in_trans = skip_next_in_trans;
         }
     }
 
-    const auto finalSize = m_layers.back().outputs * one_plane;
-    queue.enqueueReadBuffer(inBuffer, CL_FALSE, 0, finalSize, output.data());
+    auto pinnedOutBufferHost = queue.enqueueMapBuffer(
+        opencl_thread_data.m_pinnedOutBuffer, CL_FALSE,
+        CL_MAP_READ, 0, finalSize);
 
     std::lock_guard<std::mutex> lock(m_queue_finish_mutex);
     queue.finish();
+
```

Comment:
The original intention was that we unlock m_queue_finish_mutex right after queue.finish() -  the mutex was there for preventing too many threads from waiting on the finish() call (which is a busy wait on many OpenCL implementations)